docs/reST/ref/display.rst

```diff
@@ -42,6 +42,42 @@ user adjusts the window dimensions. Hardware displays that draw direct to the
 screen will get ``pygame.VIDEOEXPOSE`` events when portions of the window must
 be redrawn.
 
+In pygame 2 (when built with SDL 2), there is a new type of event called
+``pygame.WINDOWEVENT`` that is meant to replace all window related events
+like ``pygame.VIDEORESIZE``, ``pygame.VIDEOEXPOSE`` and ``pygame.ACTIVEEVENT``.
+
+While SDL has removed support for these old events, pygame is still supporting
+them for backwards compatability.
+The new events of type ``pygame.WINDOWEVENT`` have an ``event`` attribute that 
+can take the following values.
+
+::
+
+   Value of event attribute      Short description
+         
+   WINDOWEVENT_SHOWN            Window became shown
```

Comment:
The docs are good. 

So good, I think we need to look again at WINDOWEVENT. I don't think it recieved enough attention when we added it.

Doing a repr(event) on it, shows stuff like <WindowEvent 'event': 3>, which is meaningless.

Also it's quite verbose to use:
` event.type == pygame.WINDOWEVENT and event.event == pygame.WINDOWEVENT_HIDDEN:`
VS using attributes:
`if event.type == pygame.WINDOWEVENT and event.hidden:`
VS using a separate event type for each one:
`if event.type == pygame.WINDOWEVENT_HIDDEN:`

Having to handle the WINDOWEVENT differently than all the other types is an extra load on the user.

Reply:
Hmm, that would involve a lot of code change though

Reply:
Added an issue about this here: https://github.com/pygame/pygame/issues/2214